### PR TITLE
Improve and update the documentation

### PR DESCRIPTION
**Pull request by Kriskras99**
I noticed various things in the docs that could be improved, feel free to cherry-pick the changes you like.

Both README.md and the lib.rs documentation were missing features the other had. 
I've merged the two and updated where necessary, and made it so that lib.rs includes the README.md content.
I've also made various small fixes in various places.

### CODE REVIEW

**Review by Licenser — APPROVED** [review]
This is awesome, thank you!

File: README.md [hunk 1]
```diff
@@ -10,89 +10,170 @@
...
+let v: simd_json::BorrowedValue = simd_json::to_borrowed_value(&mut d).unwrap();
+```
+
+**Owned Values**
+
+```rust
+use simd_json;
+let mut d = br#"{"some": ["key", "value", 2]}"#.to_vec();
+let v: simd_json::OwnedValue = simd_json::to_owned_value(&mut d).unwrap();
+```
```
[the remark this answers is not in the document]
Comment (Kriskras99) [reply README.md:166]: I'll add one tonight

[the remark this answers is not in the document]
Comment (Licenser) [reply README.md:166]: I'm happy to merge it as is and do that in another PR, whatever you prefer :)

[the remark this answers is not in the document]
Comment (Kriskras99) [reply README.md:166]: I've added an example, I hope it's correct. It does pas the doctests.

**Review by Licenser — APPROVED** [review]
Thank you so much :D, this is great!

**Pull request merged**